From 0b2a293a76b1e1e7d021ad3b13c77c2660597f7d Mon Sep 17 00:00:00 2001
From: Robert Wiegand <qa2113@motorola.com>
Date: Tue, 17 May 2016 12:18:48 -0500
Subject: [PATCH 872/959] IKLOCSEN-2684 drives: misc: stml0xx: Motion and
 Stationary Detect

Add support for the new Motion and Stationary Detect sensors
that will be added in the Android N release.

Change-Id: I69a29cab5863eb154fcac9dc7192c496cd819adc
Signed-off-by: Robert Wiegand <qa2113@motorola.com>
Reviewed-on: http://gerrit.mot.com/857514
SME-Granted: SME Approvals Granted
SLTApproved: Slta Waiver <sltawvr@motorola.com>
Submit-Approved: Jira Key <jirakey@motorola.com>
Tested-by: Jira Key <jirakey@motorola.com>
Reviewed-by: Eric Hefner <ehefner@motorola.com>
Reviewed-by: Daniel Moore <dtmoore@motorola.com>
Reviewed-by: Jason Jae Hyun Koh <jasonkoh@motorola.com>
Reviewed-by: Igor Kovalenko <igork@motorola.com>
---
 drivers/misc/stml0xx/stml0xx_wake_irq.c | 10 ++++++++++
 include/uapi/linux/stml0xx.h            |  8 +++++---
 2 files changed, 15 insertions(+), 3 deletions(-)

diff --git a/drivers/misc/stml0xx/stml0xx_wake_irq.c b/drivers/misc/stml0xx/stml0xx_wake_irq.c
index aad59b314e9..675dbab0c11 100644
--- a/drivers/misc/stml0xx/stml0xx_wake_irq.c
+++ b/drivers/misc/stml0xx/stml0xx_wake_irq.c
@@ -435,6 +435,16 @@ void stml0xx_irq_wake_work_func(struct work_struct *work)
 		dev_dbg(&stml0xx_misc_data->spi->dev, "Glance Gesture=%d\n",
 				SH_TO_H16(buf + WAKE_IRQ_IDX_GLANCE));
 	}
+	if (irq_status & M_MOTION_DETECT) {
+		stml0xx_as_data_buffer_write(ps_stml0xx, DT_MOTION_DETECT,
+				NULL, 0, 0, stm_ws->ts_ns);
+		dev_dbg(&stml0xx_misc_data->spi->dev, "Motion Detect");
+	}
+	if (irq_status & M_STATIONARY_DETECT) {
+		stml0xx_as_data_buffer_write(ps_stml0xx, DT_STATIONARY_DETECT,
+				NULL, 0, 0, stm_ws->ts_ns);
+		dev_dbg(&stml0xx_misc_data->spi->dev, "Stationary Detect");
+	}
 	if (irq2_status & M_MMOVEME) {
 		unsigned char status;
 
diff --git a/include/uapi/linux/stml0xx.h b/include/uapi/linux/stml0xx.h
index 6dab07c4a89..25f1872f67d 100644
--- a/include/uapi/linux/stml0xx.h
+++ b/include/uapi/linux/stml0xx.h
@@ -217,8 +217,8 @@
 #define M_UPDATE_GYRO_CAL       0x020000
 #define M_GLANCE                0x040000
 /*#define M_UNUSED              0x080000*/
-/*#define M_UNUSED              0x100000*/
-/*#define M_UNUSED              0x200000*/
+#define M_MOTION_DETECT         0x100000
+#define M_STATIONARY_DETECT     0x200000
 /*#define M_UNUSED              0x400000*/
 /*#define M_UNUSED              0x800000*/
 
@@ -321,7 +321,9 @@ enum STML0XX_data_types {
 	DT_STEP_COUNTER,
 	DT_STEP_DETECTOR,
 	DT_GLANCE,
-	DT_ACCEL_CAL
+	DT_ACCEL_CAL,
+	DT_MOTION_DETECT,
+	DT_STATIONARY_DETECT
 };
 
 enum {
-- 
2.11.0

